test/cpp/api/modules.cpp

```diff
@@ -108,6 +108,20 @@ TEST_F(ModulesTest, MaxPool1d) {
   ASSERT_EQ(y.sizes(), torch::IntArrayRef({1, 1, 2}));
 }
 
+TEST_F(ModulesTest, MaxPool1dReturnIndices) {
+  MaxPool1d model(MaxPool1dOptions(3).stride(2));
+  auto x = torch::ones({1, 1, 5}, torch::requires_grad());
+  torch::Tensor y, indices;
+  std::tie(y, indices) = model->forward_with_indices(x);
+
+  ASSERT_EQ(y.ndimension(), 3);
+  ASSERT_TRUE(torch::allclose(y, torch::ones({1, 1 ,2})));
+  ASSERT_EQ(y.sizes(), torch::IntArrayRef({1, 1, 2}));
```

Comment:
Does this actually work? Can one construct an ArrayRef for a temporary variable and have it stick around?

Reply:
Based on https://stackoverflow.com/a/2506800, temporary objects are destroyed at the end of the full expression they're part of, so I think this should work.

Reply:
I see that this already exists in the code already and it's never triggered an asan check so it should be fine.

Okay, so it sounds like under the hood IntArrayRef is storing some const references?

Reply:
It's storing a const pointer to the data buffer:
https://github.com/pytorch/pytorch/blob/3f72bcfcaacadfbbfb40e1cc32fc15fdec26bca5/c10/util/ArrayRef.h#L52
I think it only works here because we know that the temporary variable won't be destroyed before the assert check is done, otherwise we would need to change it to something like `ASSERT_EQ(y.sizes(), std::vector({1, 1, 2}));`